mcu/atmega-hal/src/usart_spi.rs

```diff
@@ -0,0 +1,104 @@
+//! USART MSPIM implimentations
+//!
+//! The following list details how many USARTs and if the USARTs support MSPIM for each board choosable.
+//!
+//! | Board | USARTs | SPI |
+//! |-------|--------|-----|
+//! | `atmega48p` | 1 | Yes |
+//! | `atmega164pa`| 2 | Yes |
+//! | `atmega168` | 1 | Yes |
+//! | `atmega328p` | 1 | Yes |
+//! | `atmega328pb` | 1 | Yes |
+//! | `atmega32a` | 1 | No |
+//! | `atmega32u4` | 1 | Yes |
+//! | `atmega2560` | 4 | Yes |
+//! | `atmega128a` | 2 | No |
+//! | `atmega1280` | 4 | Yes |
+//! | `atmega1284p` | 2 | Yes |
+//! | `atmega8` | 1 | No |
+
+// Supress warning because it doesn't recognise us using it in macros properly.
+#[allow(unused_imports)]
+use crate::port;
+
+#[cfg(any(feature = "atmega1280", feature = "atmega2560"))]
+avr_hal_generic::add_usart_spi! {
+    hal: crate::Atmega,
+    peripheral: crate::pac::USART0,
+    register_suffix: 0,
+    sclk: port::PE2,
+    mosi: port::PE1,
+    miso: port::PE0,
+    cs: port::Dynamic,
```

Comment:
Hm, this is quite a hack.  We really don't need the `cs` dance for MSPIM so it would be nice to somehow wrangle the SPI module to not require a CS pin for MSPIM...

Reply:
This was done because I impliment SpiOps to cheap out on work, and not rewrite everything.

I can either reimplement a new spiops, or, force the use of a DummyPin internally (so users don’t have to worry about it) via the DummyPin crate (I believe that’s its name)

Reply:
I would be fine with defining a `UsartSpiDummyCs` pin type that needs to be passed to `Spi::new()` to sidestep this.  Pulling in an external crate for this is not a good idea, I think.

Reply:
An alternative to the Dummy CS Pin (and what I believe my original thought process was), is how it is, where it's dynamic, meaning the CS can be handled by the driver in some which way, by manual toggle instead of SPI handling it.